dispatchers.go

```diff
@@ -313,6 +315,17 @@ func workdir(b *Builder, args []string, attributes map[string]bool, flagArgs []s
 	return nil
 }
 
+func parseCommandForHeredoc(firstline string) string {
+	// Expected format `RUN <str> <<EOF`
+	if strings.HasPrefix(firstline, "RUN ") {
+		firstline = strings.TrimPrefix(firstline, "RUN ")
+		if strings.HasSuffix(firstline, "<<EOF") {
```

Comment:
Should we support <<eof as well?  Just add a ToUpper to Firstline?

Reply:
Or should we generalize this totally and allow `<< foobar`

Reply:
Why is this only RUN, shouldn't this handle COPY and ADD?
Since RUN can be lowercase, this should also use ToUpper.

Reply:
@rhatdan Yes PR is still WIP, I need to extended this to `COPY` and `ADD` and yes this should be not restricted to `<<EOF` it should work with any identifier. The code is pretty much in draft and still under progress.

Reply:
Extended for `ADD` and `COPY` as well.